### PR TITLE
Update erc20 precompile transfer to use extrinsic

### PR DESCRIPTION
**Pull request by KarishmaBothara**
**PR checklist**  
Currently some precompiles call the trait internally rather than the extrinsic directly, such as ERC20-transfer. This means that if we block the assets-transfer extrinsic, the precompile will still function normally. If we want to be able to block precompiles with the maintenance-mode block_call extrinsic, these precompiles will need to change their implementation.

We can block precompile addresses from executing which can achieve similar results, however this may not give us the full flexibility we require

I have attached the precompile list which calls trait internally in the ticket

Update ERC20 precompile in this PR to use extrinsic over trait

Closes https://futureverse.atlassian.net/browse/TRN-166

- [x] Describe the added/changed/fixed/removed behaviour

### CODE REVIEW

File: evm-precompiles/erc20/src/lib.rs [hunk 1]
```diff
@@ -277,19 +279,26 @@ where
 		handle.record_cost(Runtime::GasWeightMapping::weight_to_gas(
 			<Runtime as pallet_assets::Config>::WeightInfo::transfer(),
 		))?;
-		let _ = <pallet_assets_ext::Pallet<Runtime> as Transfer<Runtime::AccountId>>::transfer(
```
Comment (JasonTulp) [inline evm-precompiles/erc20/src/lib.rs:280]: We can remove the handle.record_cost above this which manually records the cost of the transfer extrinsic. By using try_dispatch we no longer need this

File: evm-precompiles/erc20/src/lib.rs [hunk 2]
```diff
@@ -339,16 +348,19 @@ where
 			handle.record_cost(Runtime::GasWeightMapping::weight_to_gas(
 				<Runtime as pallet_assets::Config>::WeightInfo::transfer(),
 			))?;
-			let _ = <pallet_assets_ext::Pallet<Runtime> as Transfer<Runtime::AccountId>>::transfer(
```
Comment (JasonTulp) [inline evm-precompiles/erc20/src/lib.rs:342]: Can remove the manual record_cost above this too

File: e2e/test/ERC20/Erc20.TxCosts.test.ts [hunk 3]
```diff
@@ -216,7 +216,7 @@ describe("ERC20 Gas Estimates", function () {
     const extrinsicFeeCost = balanceBefore.sub(balanceAfter);
     const extrinsicGasScaled = await getScaledGasForExtrinsicFee(provider, extrinsicFeeCost);
 
-    expect(precompileGasEstimate).to.be.lessThan(contractGasEstimate);
+    expect(precompileGasEstimate).to.be.greaterThan(contractGasEstimate);
```
Comment (JasonTulp) [inline e2e/test/ERC20/Erc20.TxCosts.test.ts:219]: Hopefully once the extra gas is removed this will go back to lessThan

**Review by JasonTulp — COMMENTED** [review]
Looks good! Just need to remove the double gas calculation :)

File: e2e/test/ERC20/Erc20.TxCosts.test.ts (hunk 3 above)
Comment (KarishmaBothara) [reply e2e/test/ERC20/Erc20.TxCosts.test.ts]: thanks.. yes, you were right.. it does solves this issue.. :)

File: evm-precompiles/erc20/src/lib.rs [hunk 4]
```diff
@@ -274,22 +276,25 @@ where
...
-		))?;
-		let _ = <pallet_assets_ext::Pallet<Runtime> as Transfer<Runtime::AccountId>>::transfer(
-			asset_id,
-			&origin,
-			&to.clone().into(),
-			amount,
-			false,
-		)
-		.map_err(|e| revert(alloc::format!("ERC20: Dispatched call failed with error: {:?}", e)))?;
+		let destination = to.clone().into();
```
Comment (zees-dev) [inline evm-precompiles/erc20/src/lib.rs:279]: not sure if we need rhe `.into()` here - as we dont explicitly define the converted type
```suggestion
		let destination = to.clone();
```

File: e2e/.yarn/install-state.gz
Comment (zees-dev) [inline e2e/.yarn/install-state.gz:1]: remove this file? not sure why this is committed in.

**Review by zees-dev — CHANGES_REQUESTED** [review]
LGTM, minor changes required.
Also CI needs to be fixed/passing.

File: evm-precompiles/erc20/src/lib.rs (hunk 4 above)
Comment (KarishmaBothara) [reply evm-precompiles/erc20/src/lib.rs]: this doesn't work without `.into()`
```
--> evm-precompiles/erc20/src/lib.rs:283:18
    |
283 |                 destination: to.clone(),
    |                              ^^^^^^^^^^ expected associated type, found `H160`
    |
    = note: expected associated type `<Runtime as frame_system::Config>::AccountId`
                        found struct `H160`
    = help: consider constraining the associated type `<Runtime as frame_system::Config>::AccountId` to `H160` or calling a method that returns `<Runtime as frame_system::Config>::AccountId`
    = note: for more information, visit https://doc.rust-lang.org/book/ch19-03-advanced-traits.html
help: call `Into::into` on this expression to convert `H160` into `<Runtime as frame_system::Config>::AccountId`
    |
283 |                 destination: to.clone().into(),
    |                                        +++++++
```

Comment (zees-dev) [reply evm-precompiles/erc20/src/lib.rs]: Ah interesting, ok ignore the comment then 👍

**Review by zees-dev — APPROVED** [review]

**Pull request merged**